### ISSUE TITLE
MilStdSymbol modifiers not getting updated for Feature updates

### ISSUE THREAD

**Issue by michaeltk3** [missioncommand/emp3-android#285]
When updating an existing MilStdSymbol's unique designator 1 modifier, the updated value is not reflected on the map with the graphic. The MilStdSymbol that we have seen this issue with is Assembly Area. There may be others but that certainly doesn't work for us. So if I plot an Assembly Area with a unique designator 1 modifier equal to "Test", I see "Test" displayed as part of the graphic on the map. If I then update the unique designator 1 modifier to "Update", the "Test" text on the graphic is not replaced and continues to display.

The mechanism we're using to update an existing feature is as follows:
IFeature existingFeature = ...
IFeature updatedFeature = ...
existingFeature.apply(updatedFeature);

If this isn't the proper procedure for updating an existing feature on the map, then please advise. Thanks

**Comment by rsupnekar** [comment]
Try this:

MilStdSymbol symbol = (MilStdSymbol)existingFeature;
symbol.setModifier(IGeoMilSymbol.Modifier.UNIQUE_DESIGNATOR_1, "your text here");

**Comment by michaeltk3** [comment]
So in our case we're translating between EMP and another data model. Symbol updates that come into our app from external sources are represented in a different data model. We have a library that translates this other data representation to an EMP Feature. We want to update the existing EMP Feature without having to manually copy over every field. Was hoping that there was an easier way then that. Both the existing feature and the updated feature contain the same GeoId. Is there a way to tell EMP to replace the existing feature with a new one? I want to avoid having to remove the existing one from the map and then adding the new one. We had been doing that but it creates other issues for us.

**Comment by rsupnekar** [comment]
You are essentially adding a new feature.  Why not delete the old feature and add the new feature?

**Comment by rsupnekar** [comment]
In other words, what's the issue?

**Comment by rsupnekar** [comment]
In the apply(Object) method, the Object is a userContext, not a feature.  The userContext is in there for a user to filter out things they don't want to listen to.  The apply method works by applying the changes you made via other calls.

**Comment by rsupnekar** [comment]
By "things", I mean events such as your own events.  If you want to filter out events, you put in a string key that identifies your own app and when you get that key in an event, don't process the event.

**Comment by michaeltk3** [comment]
We had been doing just that but ran into an issue related to a properties-like dialog the user sees when they click on the feature. We have listener that disposes the dialog if the feature is removed from the map. This way the user isn't looking at a dialog that is for a non-existent feature. Problem was that one feature was getting updated once a second and so as soon as the user opened the dialog, the listener would detect that the feature was removed and dispose the dialog right away.

Thanks for explanation of the apply(Object) method. We didn't have a good understanding of that one. Prior to using that approach we had tried adding the updated feature (with the same GeoId as the existing one). This approach didn't remove the existing one. EMP almost handled this case. The only issue I was seeing was that some callbacks from EMP had references to the existing feature and not the updated one after that.
